app/lib/actions/pulp3/orchestration/repository/generate_metadata.rb

```diff
@@ -5,11 +5,12 @@ module Repository
         class GenerateMetadata < Pulp3::Abstract
           def plan(repository, smart_proxy, options = {})
             options[:contents_changed] = (options && options.key?(:contents_changed)) ? options[:contents_changed] : true
+            publication_content_type = !(repository.content_type == "ansible_collection")
```

Comment:
could we add some smarts into the repository type registration to know about this?  Maybe something like:

```
Katello::RepositoryTypeManager.register(::Katello::Repository::FILE_TYPE) do
  allow_creation_by_user true
  service_class Katello::Pulp::Repository::File
  pulp3_service_class Katello::Pulp3::Repository::File
  pulp3_plugin 'pulp_file'
  pulp3_uses_publications

  content_type Katello::FileUnit, :pulp2_service_class => ::Katello::Pulp::FileUnit, :pulp3_service_class => ::Katello::Pulp3::FileUnit, :user_removable => true
end

```

to indicate that file type uses publications (and anything that doesn't declare that, does not)